components/encryption/src/crypter.rs

```diff
@@ -197,6 +198,81 @@ pub fn verify_encryption_config(method: EncryptionMethod, key: &[u8]) -> Result<
     Ok(())
 }
 
+// Encrypted is a newtype used to mark data as encrypted
+// It requires the vec to be non-empty
+#[derive(PartialEq, Clone, Debug)]
+pub struct Encrypted(Vec<u8>);
```

Comment:
how about `EncryptedKey`, to be consistent with `PlainKey`?

Reply:
PlainKey differentiates a key from other unencrypted data. `Encrypted` can be used to mark different types of data (a key or a payload) as encrypted. I could see this developing into `Key<T>` where `T` could be `Encrypted` or not.

Reply:
I don't see the code being more clear after introducing this type. It doesn't serve any functional purpose, and don't make the logic more clear.

Reply:
I think I agree with you because we already have an EncryptedContent type in use.